## Load Dataset


In [ ]:
from datasets import load_dataset
# Load our data
data = load_dataset("rotten_tomatoes")
data

/usr/local/lib/python3.12/dist-packages/huggingface_hub/utils/_auth.py:94: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


README.md: 0.00B [00:00, ?B/s]

train.parquet:   0%|          | 0.00/699k [00:00<?, ?B/s]

validation.parquet:   0%|          | 0.00/90.0k [00:00<?, ?B/s]

test.parquet:   0%|          | 0.00/92.2k [00:00<?, ?B/s]

Generating train split:   0%|          | 0/8530 [00:00<?, ? examples/s]

Generating validation split:   0%|          | 0/1066 [00:00<?, ? examples/s]

Generating test split:   0%|          | 0/1066 [00:00<?, ? examples/s]

DatasetDict({
    train: Dataset({
        features: ['text', 'label'],
        num_rows: 8530
    })
    validation: Dataset({
        features: ['text', 'label'],
        num_rows: 1066
    })
    test: Dataset({
        features: ['text', 'label'],
        num_rows: 1066
    })
})

In [ ]:
data["train"][0,-1]

{'text': ['the rock is destined to be the 21st century\'s new " conan " and that he\'s going to make a splash even greater than arnold schwarzenegger , jean-claud van damme or steven segal .',
  'things really get weird , though not particularly scary : the movie is all portent and no content .'],
 'label': [1, 0]}

## Approach 1 - Task Specific

In [ ]:
from transformers import pipeline
# Path to our HF model
model_path = "cardiffnlp/twitter-roberta-base-sentiment-latest"
# Load model into pipeline
pipe = pipeline(
model=model_path,
tokenizer=model_path,
return_all_scores=
True
,
device="cuda:0"
)

config.json:   0%|          | 0.00/929 [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/501M [00:00<?, ?B/s]

Some weights of the model checkpoint at cardiffnlp/twitter-roberta-base-sentiment-latest were not used when initializing RobertaForSequenceClassification: ['roberta.pooler.dense.bias', 'roberta.pooler.dense.weight']
- This IS expected if you are initializing RobertaForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing RobertaForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).


model.safetensors:   0%|          | 0.00/501M [00:00<?, ?B/s]

vocab.json: 0.00B [00:00, ?B/s]

merges.txt: 0.00B [00:00, ?B/s]

special_tokens_map.json:   0%|          | 0.00/239 [00:00<?, ?B/s]

Device set to use cuda:0
/usr/local/lib/python3.12/dist-packages/transformers/pipelines/text_classification.py:111: UserWarning: `return_all_scores` is now deprecated,  if want a similar functionality use `top_k=None` instead of `return_all_scores=True` or `top_k=1` instead of `return_all_scores=False`.
  warnings.warn(


In [ ]:
import numpy as np
from tqdm import tqdm
from transformers.pipelines.pt_utils import KeyDataset
# Run inference
y_pred = []
for output in tqdm(pipe(KeyDataset(data["test"], "text")), total=len(data["test"])):
  negative_score = output[0]["score"]
  positive_score = output[2]["score"]
  assignment = np.argmax([negative_score, positive_score])
  y_pred.append(assignment)


100%|██████████| 1066/1066 [00:11<00:00, 92.50it/s] 


In [ ]:
from sklearn.metrics import classification_report
def evaluate_performance(y_true, y_pred):
  """Create and print the classification report"""
  performance = classification_report(
  y_true, y_pred,
  target_names=["Negative Review", "Positive Review"])
  print(performance)

In [ ]:
evaluate_performance(data["test"]["label"], y_pred)

                 precision    recall  f1-score   support

Negative Review       0.76      0.88      0.81       533
Positive Review       0.86      0.72      0.78       533

       accuracy                           0.80      1066
      macro avg       0.81      0.80      0.80      1066
   weighted avg       0.81      0.80      0.80      1066



## Approach 2 - Using Embedding

In [ ]:
from sentence_transformers import SentenceTransformer
# Load model
model = SentenceTransformer("jeongseokoh/sentence-transformers-all-mpnet-base-v2-contrastive")
# Convert text to embeddings
train_embeddings = model.encode(data["train"]["text"], show_progress_bar=True)
test_embeddings = model.encode(data["test"]["text"], show_progress_bar=True)

config.json:   0%|          | 0.00/545 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/438M [00:00<?, ?B/s]

tokenizer_config.json: 0.00B [00:00, ?B/s]

vocab.txt: 0.00B [00:00, ?B/s]

tokenizer.json: 0.00B [00:00, ?B/s]

special_tokens_map.json:   0%|          | 0.00/964 [00:00<?, ?B/s]

Batches:   0%|          | 0/267 [00:00<?, ?it/s]

Batches:   0%|          | 0/34 [00:00<?, ?it/s]

In [ ]:
train_embeddings[0].shape

(768,)

In [ ]:
from sklearn.linear_model import LogisticRegression
# Train a logistic regression on our train embeddings
clf = LogisticRegression(random_state=42)
clf.fit(train_embeddings, data["train"]["label"])

LogisticRegression(random_state=42)

In [ ]:
y_pred = clf.predict(test_embeddings)
evaluate_performance(data["test"]["label"], y_pred)

                 precision    recall  f1-score   support

Negative Review       0.83      0.86      0.85       533
Positive Review       0.86      0.83      0.84       533

       accuracy                           0.84      1066
      macro avg       0.84      0.84      0.84      1066
   weighted avg       0.84      0.84      0.84      1066



## Approach- 3 Zero-Short.

In [ ]:
label_embeddings = model.encode(["A negative review",  "A positive review"])

In [ ]:
label_embeddings.shape

(2, 768)

In [ ]:
# from sklearn.metrics.pairwise import cosine_similarity
# text = data['test'][0]['text']
# embedding = model.encode([text])
# cosine_similarity(embedding, label_embeddings)

array([[0.47558802, 0.5142493 ]], dtype=float32)

In [ ]:
from sklearn.metrics.pairwise import cosine_similarity
# Find the best matching label for each document
sim_matrix = cosine_similarity(test_embeddings, label_embeddings)
y_pred = np.argmax(sim_matrix, axis=1)

In [ ]:
evaluate_performance(data["test"]["label"], y_pred)

                 precision    recall  f1-score   support

Negative Review       0.70      0.89      0.78       533
Positive Review       0.85      0.61      0.71       533

       accuracy                           0.75      1066
      macro avg       0.77      0.75      0.75      1066
   weighted avg       0.77      0.75      0.75      1066



## Approach 4. Generative Models

In [ ]:
# Load our model
pipe = pipeline(
"text2text-generation",
model="google/flan-t5-small",
device="cuda:0"
)

config.json: 0.00B [00:00, ?B/s]

model.safetensors:   0%|          | 0.00/308M [00:00<?, ?B/s]

generation_config.json:   0%|          | 0.00/147 [00:00<?, ?B/s]

tokenizer_config.json: 0.00B [00:00, ?B/s]

spiece.model:   0%|          | 0.00/792k [00:00<?, ?B/s]

tokenizer.json: 0.00B [00:00, ?B/s]

special_tokens_map.json: 0.00B [00:00, ?B/s]

Device set to use cuda:0


In [ ]:
prompt = "Is the following sentence positive or negative? "
data = data.map(lambda example: {"t5": prompt + example['text']})
data

Map:   0%|          | 0/8530 [00:00<?, ? examples/s]

Map:   0%|          | 0/1066 [00:00<?, ? examples/s]

Map:   0%|          | 0/1066 [00:00<?, ? examples/s]

DatasetDict({
    train: Dataset({
        features: ['text', 'label', 't5'],
        num_rows: 8530
    })
    validation: Dataset({
        features: ['text', 'label', 't5'],
        num_rows: 1066
    })
    test: Dataset({
        features: ['text', 'label', 't5'],
        num_rows: 1066
    })
})

In [ ]:
# Run inference
y_pred = []
for output in tqdm(pipe(KeyDataset(data["test"], "t5")), total=len(data["test"])):
  print(output)
  text = output[0]["generated_text"]
  print(text)
  y_pred.append(0 if text == "negative" else 1)

  0%|          | 2/1066 [00:01<13:11,  1.35it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  0%|          | 4/1066 [00:02<06:42,  2.64it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


  0%|          | 5/1066 [00:02<05:37,  3.15it/s]

[{'generated_text': 'negative'}]
negative


  1%|          | 7/1066 [00:02<04:27,  3.96it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  1%|          | 8/1066 [00:03<04:27,  3.95it/s]

[{'generated_text': 'negative'}]
negative


  1%|          | 10/1066 [00:03<03:33,  4.94it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  1%|          | 12/1066 [00:03<02:41,  6.54it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


  1%|▏         | 14/1066 [00:03<02:42,  6.49it/s]

[{'generated_text': 'positive'}]
positive


  2%|▏         | 17/1066 [00:04<02:36,  6.69it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  2%|▏         | 19/1066 [00:04<02:18,  7.57it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  2%|▏         | 23/1066 [00:04<01:49,  9.57it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


  2%|▏         | 25/1066 [00:05<01:40, 10.34it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


  3%|▎         | 29/1066 [00:05<01:38, 10.54it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


  3%|▎         | 31/1066 [00:05<01:33, 11.03it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  3%|▎         | 33/1066 [00:05<01:51,  9.26it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


  3%|▎         | 36/1066 [00:06<01:41, 10.19it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  4%|▎         | 38/1066 [00:06<02:16,  7.54it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


  4%|▍         | 40/1066 [00:06<02:25,  7.06it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  4%|▍         | 42/1066 [00:07<02:20,  7.31it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


  4%|▍         | 44/1066 [00:07<02:15,  7.56it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  4%|▍         | 46/1066 [00:07<02:29,  6.83it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


  5%|▍         | 48/1066 [00:08<02:36,  6.51it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


  5%|▍         | 50/1066 [00:08<02:16,  7.44it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  5%|▍         | 52/1066 [00:08<02:18,  7.32it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


  5%|▌         | 54/1066 [00:08<02:06,  7.98it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


  5%|▌         | 57/1066 [00:09<01:51,  9.09it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  6%|▌         | 59/1066 [00:09<01:45,  9.58it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


  6%|▌         | 61/1066 [00:09<02:04,  8.05it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  6%|▌         | 63/1066 [00:09<02:35,  6.45it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  6%|▌         | 64/1066 [00:10<03:05,  5.41it/s]

[{'generated_text': 'positive'}]
positive


  6%|▌         | 66/1066 [00:10<03:17,  5.08it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  6%|▋         | 67/1066 [00:10<03:42,  4.48it/s]

[{'generated_text': 'negative'}]
negative


  6%|▋         | 68/1066 [00:11<04:10,  3.98it/s]

[{'generated_text': 'positive'}]
positive


  7%|▋         | 70/1066 [00:11<03:45,  4.42it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  7%|▋         | 73/1066 [00:11<02:36,  6.34it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  7%|▋         | 75/1066 [00:12<02:08,  7.69it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  7%|▋         | 78/1066 [00:12<01:43,  9.50it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  8%|▊         | 80/1066 [00:12<01:45,  9.38it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  8%|▊         | 82/1066 [00:12<01:45,  9.36it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  8%|▊         | 86/1066 [00:13<01:32, 10.62it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  8%|▊         | 88/1066 [00:13<01:25, 11.47it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  9%|▊         | 92/1066 [00:13<01:19, 12.32it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  9%|▉         | 94/1066 [00:13<01:18, 12.34it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  9%|▉         | 98/1066 [00:14<01:24, 11.49it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


  9%|▉         | 100/1066 [00:14<01:22, 11.76it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 10%|▉         | 104/1066 [00:14<01:17, 12.43it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 10%|▉         | 106/1066 [00:14<01:15, 12.72it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 10%|█         | 110/1066 [00:15<01:13, 13.04it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 11%|█         | 112/1066 [00:15<01:16, 12.43it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 11%|█         | 116/1066 [00:15<01:14, 12.69it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 11%|█         | 118/1066 [00:15<01:16, 12.43it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 11%|█▏        | 122/1066 [00:16<01:16, 12.42it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 12%|█▏        | 126/1066 [00:16<01:16, 12.24it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 12%|█▏        | 128/1066 [00:16<01:19, 11.87it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 12%|█▏        | 130/1066 [00:17<02:00,  7.77it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 12%|█▏        | 133/1066 [00:17<02:33,  6.09it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 13%|█▎        | 135/1066 [00:18<02:43,  5.69it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 13%|█▎        | 136/1066 [00:18<02:50,  5.46it/s]

[{'generated_text': 'positive'}]
positive


 13%|█▎        | 138/1066 [00:18<03:01,  5.12it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 13%|█▎        | 139/1066 [00:19<03:46,  4.09it/s]

[{'generated_text': 'positive'}]
positive


 13%|█▎        | 140/1066 [00:19<03:58,  3.88it/s]

[{'generated_text': 'positive'}]
positive


 13%|█▎        | 141/1066 [00:19<04:05,  3.76it/s]

[{'generated_text': 'positive'}]
positive


 13%|█▎        | 142/1066 [00:19<04:08,  3.71it/s]

[{'generated_text': 'positive'}]
positive


 14%|█▎        | 144/1066 [00:20<03:47,  4.05it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 14%|█▎        | 146/1066 [00:20<02:52,  5.34it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 14%|█▍        | 148/1066 [00:20<02:49,  5.41it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 14%|█▍        | 150/1066 [00:21<02:38,  5.79it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 14%|█▍        | 153/1066 [00:21<02:02,  7.44it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 15%|█▍        | 155/1066 [00:21<01:40,  9.08it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 15%|█▍        | 159/1066 [00:22<01:28, 10.29it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 15%|█▌        | 161/1066 [00:22<01:22, 10.96it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 15%|█▌        | 165/1066 [00:22<01:17, 11.56it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 16%|█▌        | 167/1066 [00:22<01:15, 11.95it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 16%|█▌        | 171/1066 [00:23<01:12, 12.41it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 16%|█▌        | 173/1066 [00:23<01:10, 12.70it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 17%|█▋        | 177/1066 [00:23<01:10, 12.62it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 17%|█▋        | 179/1066 [00:23<01:10, 12.62it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 17%|█▋        | 183/1066 [00:24<01:07, 13.15it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 17%|█▋        | 185/1066 [00:24<01:05, 13.39it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 18%|█▊        | 187/1066 [00:24<01:08, 12.92it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 18%|█▊        | 189/1066 [00:24<01:43,  8.49it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 18%|█▊        | 192/1066 [00:25<02:10,  6.71it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 18%|█▊        | 193/1066 [00:25<02:19,  6.27it/s]

[{'generated_text': 'positive'}]
positive


 18%|█▊        | 195/1066 [00:26<02:40,  5.42it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 18%|█▊        | 197/1066 [00:26<02:39,  5.45it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 19%|█▊        | 198/1066 [00:26<02:27,  5.90it/s]

[{'generated_text': 'negative'}]
negative


 19%|█▊        | 199/1066 [00:26<02:35,  5.58it/s]

[{'generated_text': 'positive'}]
positive


 19%|█▉        | 200/1066 [00:26<02:46,  5.21it/s]

[{'generated_text': 'negative'}]
negative


 19%|█▉        | 202/1066 [00:27<02:45,  5.22it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 19%|█▉        | 203/1066 [00:27<02:50,  5.06it/s]

[{'generated_text': 'positive'}]
positive


 19%|█▉        | 205/1066 [00:27<02:47,  5.13it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 20%|█▉        | 208/1066 [00:28<01:54,  7.51it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 20%|█▉        | 211/1066 [00:28<01:30,  9.47it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 20%|█▉        | 213/1066 [00:28<01:25, 10.00it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 20%|██        | 215/1066 [00:28<01:19, 10.74it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 21%|██        | 219/1066 [00:29<01:21, 10.40it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 21%|██        | 221/1066 [00:29<01:17, 10.95it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 21%|██        | 225/1066 [00:29<01:17, 10.85it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 21%|██▏       | 227/1066 [00:29<01:16, 10.99it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 22%|██▏       | 231/1066 [00:30<01:09, 12.04it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 22%|██▏       | 233/1066 [00:30<01:08, 12.13it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 22%|██▏       | 237/1066 [00:30<01:08, 12.10it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 22%|██▏       | 239/1066 [00:30<01:07, 12.33it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 23%|██▎       | 243/1066 [00:31<01:09, 11.91it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 23%|██▎       | 245/1066 [00:31<01:27,  9.43it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 23%|██▎       | 247/1066 [00:31<01:30,  9.08it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 23%|██▎       | 249/1066 [00:32<01:45,  7.78it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 23%|██▎       | 250/1066 [00:32<01:51,  7.29it/s]

[{'generated_text': 'negative'}]
negative


 24%|██▎       | 251/1066 [00:32<02:23,  5.69it/s]

[{'generated_text': 'negative'}]
negative


 24%|██▎       | 252/1066 [00:33<03:11,  4.24it/s]

[{'generated_text': 'negative'}]
negative


 24%|██▎       | 253/1066 [00:33<03:41,  3.67it/s]

[{'generated_text': 'positive'}]
positive


 24%|██▍       | 256/1066 [00:33<02:45,  4.91it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 24%|██▍       | 258/1066 [00:34<02:38,  5.09it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 24%|██▍       | 260/1066 [00:34<02:41,  5.00it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 25%|██▍       | 262/1066 [00:35<02:32,  5.29it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 25%|██▍       | 263/1066 [00:35<02:36,  5.12it/s]

[{'generated_text': 'positive'}]
positive


 25%|██▍       | 265/1066 [00:35<02:42,  4.94it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 25%|██▌       | 268/1066 [00:36<01:47,  7.43it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 25%|██▌       | 269/1066 [00:36<01:56,  6.85it/s]

[{'generated_text': 'positive'}]
positive


 25%|██▌       | 271/1066 [00:36<02:21,  5.61it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 26%|██▌       | 274/1066 [00:36<01:41,  7.81it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 26%|██▌       | 277/1066 [00:37<01:25,  9.25it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 26%|██▌       | 279/1066 [00:37<01:21,  9.68it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 26%|██▋       | 281/1066 [00:37<01:28,  8.90it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 27%|██▋       | 283/1066 [00:37<01:16, 10.21it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 27%|██▋       | 287/1066 [00:38<01:07, 11.55it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 27%|██▋       | 289/1066 [00:38<01:08, 11.42it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 27%|██▋       | 293/1066 [00:38<01:11, 10.79it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 28%|██▊       | 295/1066 [00:38<01:12, 10.66it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 28%|██▊       | 299/1066 [00:39<01:06, 11.56it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 28%|██▊       | 301/1066 [00:39<01:03, 11.97it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 29%|██▊       | 305/1066 [00:39<01:05, 11.62it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 29%|██▉       | 307/1066 [00:39<01:05, 11.65it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 29%|██▉       | 309/1066 [00:40<01:02, 12.06it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 29%|██▉       | 311/1066 [00:40<01:08, 11.00it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 29%|██▉       | 314/1066 [00:40<01:31,  8.25it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 30%|██▉       | 315/1066 [00:40<01:42,  7.29it/s]

[{'generated_text': 'positive'}]
positive


 30%|██▉       | 316/1066 [00:41<01:54,  6.53it/s]

[{'generated_text': 'positive'}]
positive


 30%|██▉       | 317/1066 [00:41<02:05,  5.96it/s]

[{'generated_text': 'positive'}]
positive


 30%|██▉       | 319/1066 [00:41<02:18,  5.39it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 30%|███       | 320/1066 [00:41<02:18,  5.37it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 30%|███       | 323/1066 [00:42<02:19,  5.34it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 30%|███       | 325/1066 [00:42<02:28,  5.01it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 31%|███       | 327/1066 [00:43<01:41,  7.27it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 31%|███       | 328/1066 [00:43<01:45,  6.97it/s]

[{'generated_text': 'positive'}]
positive


 31%|███       | 329/1066 [00:43<02:12,  5.57it/s]

[{'generated_text': 'negative'}]
negative


 31%|███       | 331/1066 [00:43<02:21,  5.18it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 31%|███       | 332/1066 [00:44<02:42,  4.51it/s]

[{'generated_text': 'negative'}]
negative


 31%|███▏      | 334/1066 [00:44<02:25,  5.05it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 32%|███▏      | 336/1066 [00:44<02:16,  5.35it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 32%|███▏      | 338/1066 [00:45<02:07,  5.71it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 32%|███▏      | 340/1066 [00:45<02:00,  6.01it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 32%|███▏      | 342/1066 [00:45<01:59,  6.05it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 32%|███▏      | 344/1066 [00:46<02:05,  5.76it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 32%|███▏      | 346/1066 [00:46<01:35,  7.56it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 33%|███▎      | 350/1066 [00:46<01:12,  9.81it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 33%|███▎      | 352/1066 [00:46<01:07, 10.59it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 33%|███▎      | 356/1066 [00:47<01:04, 11.05it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 34%|███▎      | 358/1066 [00:47<01:02, 11.40it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 34%|███▍      | 362/1066 [00:47<01:02, 11.18it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 34%|███▍      | 364/1066 [00:48<01:00, 11.53it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 35%|███▍      | 368/1066 [00:48<00:56, 12.42it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 35%|███▍      | 370/1066 [00:48<00:58, 11.97it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 35%|███▌      | 374/1066 [00:48<01:06, 10.44it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 35%|███▌      | 376/1066 [00:49<01:01, 11.25it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 35%|███▌      | 378/1066 [00:49<01:07, 10.22it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 36%|███▌      | 380/1066 [00:49<01:22,  8.28it/s]

[{'generated_text': 'positive'}]
positive


 36%|███▌      | 382/1066 [00:50<01:43,  6.60it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 36%|███▌      | 384/1066 [00:50<01:46,  6.37it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 36%|███▌      | 385/1066 [00:50<01:47,  6.31it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 36%|███▋      | 387/1066 [00:50<01:34,  7.22it/s]

[{'generated_text': 'negative'}]
negative


 36%|███▋      | 389/1066 [00:51<01:41,  6.66it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 37%|███▋      | 391/1066 [00:51<01:53,  5.95it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 37%|███▋      | 393/1066 [00:51<01:34,  7.11it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 37%|███▋      | 394/1066 [00:51<01:32,  7.30it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 37%|███▋      | 397/1066 [00:52<01:26,  7.75it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 38%|███▊      | 400/1066 [00:52<01:20,  8.31it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 38%|███▊      | 403/1066 [00:52<01:07,  9.83it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 38%|███▊      | 405/1066 [00:53<01:13,  8.93it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 38%|███▊      | 408/1066 [00:53<01:03, 10.33it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 38%|███▊      | 410/1066 [00:53<01:01, 10.70it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 39%|███▉      | 414/1066 [00:53<00:55, 11.80it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 39%|███▉      | 416/1066 [00:54<00:53, 12.14it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 39%|███▉      | 420/1066 [00:54<00:55, 11.54it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 40%|███▉      | 422/1066 [00:54<00:57, 11.27it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 40%|███▉      | 426/1066 [00:54<00:54, 11.75it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 40%|████      | 428/1066 [00:55<00:53, 11.95it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 41%|████      | 432/1066 [00:55<00:53, 11.85it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 41%|████      | 434/1066 [00:55<00:55, 11.41it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 41%|████      | 438/1066 [00:55<00:51, 12.17it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 41%|████▏     | 440/1066 [00:56<00:50, 12.36it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 41%|████▏     | 442/1066 [00:56<00:51, 12.03it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 42%|████▏     | 444/1066 [00:56<01:10,  8.81it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 42%|████▏     | 446/1066 [00:57<01:49,  5.67it/s]

[{'generated_text': 'positive'}]
positive


 42%|████▏     | 447/1066 [00:57<02:07,  4.86it/s]

[{'generated_text': 'positive'}]
positive


 42%|████▏     | 448/1066 [00:57<02:24,  4.27it/s]

[{'generated_text': 'positive'}]
positive


 42%|████▏     | 449/1066 [00:58<02:37,  3.92it/s]

[{'generated_text': 'positive'}]
positive


 42%|████▏     | 450/1066 [00:58<02:44,  3.74it/s]

[{'generated_text': 'positive'}]
positive


 42%|████▏     | 451/1066 [00:58<02:49,  3.62it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 42%|████▏     | 453/1066 [00:59<02:20,  4.37it/s]

[{'generated_text': 'negative'}]
negative


 43%|████▎     | 454/1066 [00:59<02:21,  4.32it/s]

[{'generated_text': 'negative'}]
negative


 43%|████▎     | 455/1066 [00:59<02:25,  4.19it/s]

[{'generated_text': 'negative'}]
negative


 43%|████▎     | 456/1066 [00:59<02:24,  4.22it/s]

[{'generated_text': 'positive'}]
positive


 43%|████▎     | 459/1066 [01:00<01:39,  6.08it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 43%|████▎     | 461/1066 [01:00<01:31,  6.61it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 43%|████▎     | 463/1066 [01:00<01:20,  7.53it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 44%|████▎     | 466/1066 [01:01<01:09,  8.65it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 44%|████▍     | 468/1066 [01:01<00:58, 10.28it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 44%|████▍     | 471/1066 [01:01<01:01,  9.64it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 45%|████▍     | 475/1066 [01:02<00:56, 10.44it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 45%|████▍     | 477/1066 [01:02<00:55, 10.56it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 45%|████▍     | 479/1066 [01:02<00:52, 11.16it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 45%|████▌     | 483/1066 [01:02<00:53, 10.90it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 45%|████▌     | 485/1066 [01:02<00:52, 11.00it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 46%|████▌     | 489/1066 [01:03<00:48, 11.95it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 46%|████▌     | 491/1066 [01:03<00:46, 12.35it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 46%|████▋     | 495/1066 [01:03<00:47, 11.92it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 47%|████▋     | 498/1066 [01:04<01:05,  8.65it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 47%|████▋     | 502/1066 [01:04<00:54, 10.31it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 47%|████▋     | 504/1066 [01:04<00:52, 10.63it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 48%|████▊     | 508/1066 [01:05<00:53, 10.42it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 48%|████▊     | 510/1066 [01:05<01:04,  8.65it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 48%|████▊     | 512/1066 [01:05<00:57,  9.60it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 48%|████▊     | 514/1066 [01:05<00:58,  9.47it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 48%|████▊     | 517/1066 [01:06<01:19,  6.94it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 49%|████▊     | 519/1066 [01:06<01:26,  6.35it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 49%|████▉     | 521/1066 [01:07<01:29,  6.09it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 49%|████▉     | 523/1066 [01:07<01:37,  5.56it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 49%|████▉     | 524/1066 [01:07<01:36,  5.61it/s]

[{'generated_text': 'positive'}]
positive


 49%|████▉     | 527/1066 [01:08<01:17,  6.99it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 50%|████▉     | 529/1066 [01:08<01:03,  8.47it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 50%|████▉     | 532/1066 [01:08<01:00,  8.79it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 50%|█████     | 534/1066 [01:08<01:08,  7.78it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 50%|█████     | 535/1066 [01:09<01:19,  6.69it/s]

[{'generated_text': 'positive'}]
positive


 50%|█████     | 536/1066 [01:09<01:35,  5.55it/s]

[{'generated_text': 'negative'}]
negative


 50%|█████     | 537/1066 [01:09<02:00,  4.38it/s]

[{'generated_text': 'negative'}]
negative


 51%|█████     | 539/1066 [01:10<02:07,  4.12it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 51%|█████     | 541/1066 [01:10<01:50,  4.76it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 51%|█████     | 543/1066 [01:10<01:43,  5.06it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 51%|█████     | 545/1066 [01:11<01:22,  6.29it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 51%|█████▏    | 547/1066 [01:11<01:16,  6.80it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 52%|█████▏    | 550/1066 [01:11<01:03,  8.09it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 52%|█████▏    | 553/1066 [01:12<00:55,  9.22it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 52%|█████▏    | 554/1066 [01:12<00:54,  9.33it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 52%|█████▏    | 558/1066 [01:12<00:56,  8.98it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 53%|█████▎    | 560/1066 [01:12<00:52,  9.70it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 53%|█████▎    | 564/1066 [01:13<00:49, 10.14it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 53%|█████▎    | 566/1066 [01:13<00:45, 10.88it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 53%|█████▎    | 568/1066 [01:13<00:46, 10.70it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 54%|█████▎    | 571/1066 [01:13<00:55,  8.89it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 54%|█████▎    | 572/1066 [01:14<01:25,  5.76it/s]

[{'generated_text': 'negative'}]
negative


 54%|█████▍    | 573/1066 [01:14<01:34,  5.22it/s]

[{'generated_text': 'negative'}]
negative


 54%|█████▍    | 574/1066 [01:14<01:44,  4.72it/s]

[{'generated_text': 'negative'}]
negative


 54%|█████▍    | 575/1066 [01:15<01:52,  4.36it/s]

[{'generated_text': 'negative'}]
negative


 54%|█████▍    | 576/1066 [01:15<02:06,  3.89it/s]

[{'generated_text': 'negative'}]
negative


 54%|█████▍    | 577/1066 [01:15<01:59,  4.08it/s]

[{'generated_text': 'negative'}]
negative


 54%|█████▍    | 578/1066 [01:16<02:03,  3.94it/s]

[{'generated_text': 'negative'}]
negative


 54%|█████▍    | 580/1066 [01:16<01:45,  4.60it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 55%|█████▍    | 582/1066 [01:16<01:22,  5.84it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 55%|█████▍    | 584/1066 [01:16<01:11,  6.71it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 55%|█████▌    | 587/1066 [01:17<00:55,  8.62it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 55%|█████▌    | 589/1066 [01:17<00:58,  8.18it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 55%|█████▌    | 591/1066 [01:17<01:04,  7.35it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 56%|█████▌    | 593/1066 [01:18<01:00,  7.85it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 56%|█████▌    | 597/1066 [01:18<00:48,  9.74it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 56%|█████▌    | 599/1066 [01:18<00:49,  9.51it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 57%|█████▋    | 603/1066 [01:18<00:44, 10.51it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 57%|█████▋    | 605/1066 [01:19<00:43, 10.56it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 57%|█████▋    | 609/1066 [01:19<00:43, 10.51it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 57%|█████▋    | 611/1066 [01:19<00:42, 10.71it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 58%|█████▊    | 615/1066 [01:20<00:42, 10.57it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 58%|█████▊    | 617/1066 [01:20<00:42, 10.60it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 58%|█████▊    | 619/1066 [01:20<00:44,  9.97it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 58%|█████▊    | 621/1066 [01:20<00:45,  9.83it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 58%|█████▊    | 622/1066 [01:20<00:50,  8.84it/s]

[{'generated_text': 'negative'}]
negative


 58%|█████▊    | 623/1066 [01:21<01:00,  7.30it/s]

[{'generated_text': 'negative'}]
negative


 59%|█████▊    | 624/1066 [01:21<01:13,  6.03it/s]

[{'generated_text': 'negative'}]
negative


 59%|█████▊    | 625/1066 [01:21<01:33,  4.72it/s]

[{'generated_text': 'negative'}]
negative


 59%|█████▊    | 626/1066 [01:22<02:03,  3.55it/s]

[{'generated_text': 'negative'}]
negative


 59%|█████▉    | 627/1066 [01:22<02:28,  2.95it/s]

[{'generated_text': 'negative'}]
negative


 59%|█████▉    | 628/1066 [01:23<02:43,  2.67it/s]

[{'generated_text': 'negative'}]
negative


 59%|█████▉    | 629/1066 [01:23<02:49,  2.58it/s]

[{'generated_text': 'negative'}]
negative


 59%|█████▉    | 630/1066 [01:24<02:55,  2.49it/s]

[{'generated_text': 'negative'}]
negative


 59%|█████▉    | 632/1066 [01:24<02:09,  3.36it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 59%|█████▉    | 634/1066 [01:24<01:28,  4.90it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 60%|█████▉    | 636/1066 [01:25<01:25,  5.06it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 60%|█████▉    | 638/1066 [01:25<01:17,  5.53it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 60%|██████    | 640/1066 [01:25<01:05,  6.54it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 60%|██████    | 642/1066 [01:25<00:57,  7.38it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 60%|██████    | 644/1066 [01:26<01:01,  6.84it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 61%|██████    | 647/1066 [01:26<00:49,  8.43it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 61%|██████    | 649/1066 [01:26<00:47,  8.82it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 61%|██████    | 652/1066 [01:27<00:45,  9.11it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 61%|██████▏   | 654/1066 [01:27<00:45,  9.07it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 62%|██████▏   | 656/1066 [01:27<00:46,  8.74it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 62%|██████▏   | 659/1066 [01:27<00:49,  8.19it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 62%|██████▏   | 662/1066 [01:28<00:44,  9.03it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 62%|██████▏   | 665/1066 [01:28<00:38, 10.38it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 63%|██████▎   | 667/1066 [01:28<00:38, 10.46it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 63%|██████▎   | 669/1066 [01:28<00:40,  9.75it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 63%|██████▎   | 671/1066 [01:29<00:42,  9.23it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 63%|██████▎   | 673/1066 [01:29<00:43,  9.11it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 63%|██████▎   | 675/1066 [01:29<00:42,  9.26it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 63%|██████▎   | 676/1066 [01:29<00:49,  7.90it/s]

[{'generated_text': 'negative'}]
negative


 64%|██████▎   | 678/1066 [01:30<01:13,  5.26it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 64%|██████▍   | 680/1066 [01:30<01:21,  4.72it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 64%|██████▍   | 681/1066 [01:30<01:24,  4.56it/s]

[{'generated_text': 'negative'}]
negative


 64%|██████▍   | 682/1066 [01:31<01:35,  4.03it/s]

[{'generated_text': 'negative'}]
negative


 64%|██████▍   | 683/1066 [01:31<01:34,  4.06it/s]

[{'generated_text': 'negative'}]
negative


 64%|██████▍   | 684/1066 [01:31<01:37,  3.91it/s]

[{'generated_text': 'negative'}]
negative


 64%|██████▍   | 685/1066 [01:32<01:42,  3.73it/s]

[{'generated_text': 'negative'}]
negative


 64%|██████▍   | 687/1066 [01:32<01:29,  4.22it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 65%|██████▍   | 689/1066 [01:32<01:18,  4.80it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 65%|██████▍   | 691/1066 [01:33<01:03,  5.95it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 65%|██████▌   | 693/1066 [01:33<00:58,  6.39it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 65%|██████▌   | 695/1066 [01:33<00:47,  7.83it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 65%|██████▌   | 696/1066 [01:33<00:49,  7.42it/s]

[{'generated_text': 'negative'}]
negative


 65%|██████▌   | 697/1066 [01:34<01:07,  5.50it/s]

[{'generated_text': 'negative'}]
negative


 66%|██████▌   | 699/1066 [01:34<01:08,  5.37it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 66%|██████▌   | 701/1066 [01:34<01:05,  5.59it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 66%|██████▌   | 703/1066 [01:35<01:01,  5.88it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 66%|██████▌   | 705/1066 [01:35<00:56,  6.40it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 66%|██████▋   | 707/1066 [01:35<00:54,  6.61it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 67%|██████▋   | 709/1066 [01:36<00:56,  6.31it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 67%|██████▋   | 710/1066 [01:36<00:58,  6.11it/s]

[{'generated_text': 'negative'}]
negative


 67%|██████▋   | 712/1066 [01:36<00:56,  6.31it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 67%|██████▋   | 715/1066 [01:36<00:43,  8.09it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 67%|██████▋   | 718/1066 [01:37<00:35,  9.70it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 68%|██████▊   | 720/1066 [01:37<00:34, 10.04it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 68%|██████▊   | 722/1066 [01:37<00:37,  9.21it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 68%|██████▊   | 725/1066 [01:37<00:36,  9.28it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 68%|██████▊   | 727/1066 [01:38<00:36,  9.36it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 68%|██████▊   | 730/1066 [01:38<00:30, 10.86it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 69%|██████▊   | 732/1066 [01:38<00:40,  8.19it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 69%|██████▉   | 733/1066 [01:38<00:46,  7.14it/s]

[{'generated_text': 'positive'}]
positive


 69%|██████▉   | 734/1066 [01:39<00:55,  5.96it/s]

[{'generated_text': 'negative'}]
negative


 69%|██████▉   | 736/1066 [01:39<01:01,  5.39it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 69%|██████▉   | 737/1066 [01:39<01:08,  4.79it/s]

[{'generated_text': 'positive'}]
positive


 69%|██████▉   | 738/1066 [01:40<01:12,  4.55it/s]

[{'generated_text': 'negative'}]
negative


 69%|██████▉   | 739/1066 [01:40<01:17,  4.20it/s]

[{'generated_text': 'negative'}]
negative


 69%|██████▉   | 740/1066 [01:40<01:25,  3.82it/s]

[{'generated_text': 'negative'}]
negative


 70%|██████▉   | 741/1066 [01:41<01:31,  3.54it/s]

[{'generated_text': 'negative'}]
negative


 70%|██████▉   | 743/1066 [01:41<01:25,  3.80it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 70%|██████▉   | 745/1066 [01:41<01:04,  4.97it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 70%|███████   | 747/1066 [01:42<00:51,  6.14it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 70%|███████   | 749/1066 [01:42<00:44,  7.13it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 70%|███████   | 751/1066 [01:42<00:37,  8.41it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 71%|███████   | 753/1066 [01:42<00:37,  8.38it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 71%|███████   | 755/1066 [01:43<00:43,  7.13it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 71%|███████   | 757/1066 [01:43<00:40,  7.68it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 71%|███████   | 759/1066 [01:43<00:37,  8.10it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 71%|███████▏  | 762/1066 [01:43<00:33,  9.04it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 72%|███████▏  | 764/1066 [01:44<00:32,  9.26it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 72%|███████▏  | 766/1066 [01:44<00:32,  9.25it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 72%|███████▏  | 770/1066 [01:44<00:27, 10.70it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 72%|███████▏  | 772/1066 [01:44<00:27, 10.50it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 73%|███████▎  | 776/1066 [01:45<00:25, 11.16it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 73%|███████▎  | 778/1066 [01:45<00:26, 10.94it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 73%|███████▎  | 782/1066 [01:45<00:26, 10.52it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 74%|███████▎  | 784/1066 [01:46<00:29,  9.71it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 74%|███████▎  | 786/1066 [01:46<00:34,  8.18it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 74%|███████▍  | 788/1066 [01:46<00:35,  7.92it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 74%|███████▍  | 789/1066 [01:47<00:57,  4.78it/s]

[{'generated_text': 'negative'}]
negative


 74%|███████▍  | 790/1066 [01:47<01:15,  3.65it/s]

[{'generated_text': 'negative'}]
negative


 74%|███████▍  | 791/1066 [01:48<01:29,  3.08it/s]

[{'generated_text': 'negative'}]
negative


 74%|███████▍  | 792/1066 [01:48<01:35,  2.85it/s]

[{'generated_text': 'negative'}]
negative


 74%|███████▍  | 793/1066 [01:48<01:30,  3.02it/s]

[{'generated_text': 'negative'}]
negative


 75%|███████▍  | 795/1066 [01:49<01:20,  3.38it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 75%|███████▍  | 796/1066 [01:49<01:11,  3.76it/s]

[{'generated_text': 'negative'}]
negative


 75%|███████▍  | 798/1066 [01:50<01:05,  4.12it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 75%|███████▌  | 800/1066 [01:50<00:46,  5.71it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 75%|███████▌  | 801/1066 [01:50<00:49,  5.30it/s]

[{'generated_text': 'negative'}]
negative


 75%|███████▌  | 802/1066 [01:50<00:56,  4.71it/s]

[{'generated_text': 'negative'}]
negative


 76%|███████▌  | 805/1066 [01:51<00:43,  6.03it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 76%|███████▌  | 807/1066 [01:51<00:38,  6.77it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 76%|███████▌  | 810/1066 [01:51<00:33,  7.67it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 76%|███████▋  | 813/1066 [01:52<00:27,  9.13it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 76%|███████▋  | 815/1066 [01:52<00:28,  8.72it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 77%|███████▋  | 818/1066 [01:52<00:25,  9.72it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 77%|███████▋  | 820/1066 [01:52<00:23, 10.43it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 77%|███████▋  | 822/1066 [01:52<00:23, 10.52it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 77%|███████▋  | 825/1066 [01:53<00:24,  9.70it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 78%|███████▊  | 827/1066 [01:53<00:26,  9.13it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 78%|███████▊  | 829/1066 [01:53<00:22, 10.40it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 78%|███████▊  | 833/1066 [01:54<00:22, 10.33it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 78%|███████▊  | 835/1066 [01:54<00:23,  9.98it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 79%|███████▊  | 837/1066 [01:54<00:24,  9.52it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 79%|███████▊  | 839/1066 [01:54<00:24,  9.35it/s]

[{'generated_text': 'negative'}]
negative


 79%|███████▉  | 840/1066 [01:54<00:29,  7.79it/s]

[{'generated_text': 'negative'}]
negative


 79%|███████▉  | 841/1066 [01:55<00:34,  6.47it/s]

[{'generated_text': 'negative'}]
negative


 79%|███████▉  | 842/1066 [01:55<00:38,  5.80it/s]

[{'generated_text': 'positive'}]
positive


 79%|███████▉  | 844/1066 [01:55<00:41,  5.29it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 79%|███████▉  | 845/1066 [01:56<00:46,  4.74it/s]

[{'generated_text': 'negative'}]
negative


 79%|███████▉  | 846/1066 [01:56<01:02,  3.52it/s]

[{'generated_text': 'negative'}]
negative


 79%|███████▉  | 847/1066 [01:56<01:00,  3.59it/s]

[{'generated_text': 'negative'}]
negative


 80%|███████▉  | 849/1066 [01:57<00:55,  3.88it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 80%|███████▉  | 851/1066 [01:57<00:45,  4.68it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 80%|████████  | 853/1066 [01:58<00:38,  5.55it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 80%|████████  | 855/1066 [01:58<00:33,  6.26it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 80%|████████  | 857/1066 [01:58<00:29,  6.97it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 81%|████████  | 859/1066 [01:58<00:29,  6.92it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 81%|████████  | 862/1066 [01:59<00:24,  8.35it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 81%|████████  | 864/1066 [01:59<00:29,  6.88it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 81%|████████  | 866/1066 [01:59<00:28,  6.90it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 81%|████████▏ | 867/1066 [02:00<00:33,  5.88it/s]

[{'generated_text': 'negative'}]
negative


 82%|████████▏ | 869/1066 [02:00<00:33,  5.80it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 82%|████████▏ | 871/1066 [02:00<00:30,  6.41it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 82%|████████▏ | 873/1066 [02:01<00:33,  5.84it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 82%|████████▏ | 875/1066 [02:01<00:32,  5.96it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 82%|████████▏ | 877/1066 [02:01<00:34,  5.43it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 82%|████████▏ | 879/1066 [02:02<00:29,  6.44it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 83%|████████▎ | 881/1066 [02:02<00:26,  7.07it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 83%|████████▎ | 883/1066 [02:02<00:21,  8.36it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 83%|████████▎ | 886/1066 [02:02<00:19,  9.32it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 83%|████████▎ | 889/1066 [02:03<00:17, 10.06it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 84%|████████▎ | 892/1066 [02:03<00:16, 10.38it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 84%|████████▍ | 894/1066 [02:03<00:16, 10.46it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 84%|████████▍ | 898/1066 [02:03<00:15, 10.65it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 84%|████████▍ | 900/1066 [02:04<00:26,  6.38it/s]

[{'generated_text': 'negative'}]
negative


 85%|████████▍ | 902/1066 [02:04<00:29,  5.54it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 85%|████████▍ | 903/1066 [02:05<00:30,  5.34it/s]

[{'generated_text': 'negative'}]
negative


 85%|████████▍ | 904/1066 [02:05<00:32,  5.04it/s]

[{'generated_text': 'negative'}]
negative


 85%|████████▍ | 905/1066 [02:05<00:32,  4.90it/s]

[{'generated_text': 'negative'}]
negative


 85%|████████▍ | 906/1066 [02:05<00:37,  4.26it/s]

[{'generated_text': 'negative'}]
negative


 85%|████████▌ | 907/1066 [02:06<00:37,  4.28it/s]

[{'generated_text': 'negative'}]
negative


 85%|████████▌ | 908/1066 [02:06<00:36,  4.29it/s]

[{'generated_text': 'negative'}]
negative


 85%|████████▌ | 910/1066 [02:06<00:31,  4.88it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 85%|████████▌ | 911/1066 [02:06<00:28,  5.42it/s]

[{'generated_text': 'negative'}]
negative


 86%|████████▌ | 912/1066 [02:07<00:33,  4.66it/s]

[{'generated_text': 'negative'}]
negative


 86%|████████▌ | 914/1066 [02:07<00:32,  4.67it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 86%|████████▌ | 917/1066 [02:07<00:22,  6.68it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 86%|████████▌ | 919/1066 [02:08<00:21,  6.91it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 86%|████████▋ | 921/1066 [02:08<00:18,  7.71it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 87%|████████▋ | 925/1066 [02:08<00:14,  9.68it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 87%|████████▋ | 927/1066 [02:09<00:15,  9.19it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 87%|████████▋ | 929/1066 [02:09<00:16,  8.31it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 88%|████████▊ | 933/1066 [02:09<00:13, 10.08it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 88%|████████▊ | 936/1066 [02:09<00:11, 11.06it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 88%|████████▊ | 938/1066 [02:10<00:12, 10.24it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 88%|████████▊ | 941/1066 [02:10<00:12,  9.76it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 88%|████████▊ | 943/1066 [02:10<00:12,  9.80it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 89%|████████▊ | 945/1066 [02:10<00:12,  9.93it/s]

[{'generated_text': 'negative'}]
negative


 89%|████████▉ | 947/1066 [02:11<00:14,  8.47it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 89%|████████▉ | 949/1066 [02:11<00:14,  7.95it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 89%|████████▉ | 950/1066 [02:11<00:16,  6.86it/s]

[{'generated_text': 'positive'}]
positive


 89%|████████▉ | 951/1066 [02:11<00:20,  5.48it/s]

[{'generated_text': 'negative'}]
negative


 89%|████████▉ | 952/1066 [02:12<00:31,  3.64it/s]

[{'generated_text': 'negative'}]
negative


 89%|████████▉ | 953/1066 [02:12<00:38,  2.95it/s]

[{'generated_text': 'negative'}]
negative


 89%|████████▉ | 954/1066 [02:13<00:41,  2.72it/s]

[{'generated_text': 'negative'}]
negative


 90%|████████▉ | 955/1066 [02:13<00:44,  2.50it/s]

[{'generated_text': 'negative'}]
negative


 90%|████████▉ | 956/1066 [02:14<00:46,  2.38it/s]

[{'generated_text': 'negative'}]
negative


 90%|████████▉ | 957/1066 [02:14<00:43,  2.48it/s]

[{'generated_text': 'negative'}]
negative


 90%|████████▉ | 959/1066 [02:15<00:32,  3.28it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 90%|█████████ | 962/1066 [02:15<00:18,  5.56it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 90%|█████████ | 964/1066 [02:15<00:15,  6.68it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 91%|█████████ | 965/1066 [02:15<00:15,  6.68it/s]

[{'generated_text': 'negative'}]
negative


 91%|█████████ | 966/1066 [02:16<00:17,  5.64it/s]

[{'generated_text': 'negative'}]
negative


 91%|█████████ | 968/1066 [02:16<00:16,  6.12it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 91%|█████████ | 970/1066 [02:16<00:13,  7.16it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 91%|█████████ | 972/1066 [02:16<00:12,  7.73it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 91%|█████████▏| 974/1066 [02:17<00:11,  8.14it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 92%|█████████▏| 977/1066 [02:17<00:09,  9.53it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 92%|█████████▏| 979/1066 [02:17<00:08, 10.25it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 92%|█████████▏| 982/1066 [02:17<00:08,  9.68it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 92%|█████████▏| 984/1066 [02:18<00:08,  9.46it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 92%|█████████▏| 986/1066 [02:18<00:08,  9.38it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 93%|█████████▎| 988/1066 [02:18<00:08,  9.42it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 93%|█████████▎| 991/1066 [02:18<00:07, 10.16it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 93%|█████████▎| 993/1066 [02:19<00:07,  9.66it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 93%|█████████▎| 996/1066 [02:19<00:07,  9.61it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 94%|█████████▎| 999/1066 [02:19<00:06, 10.76it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 94%|█████████▍| 1002/1066 [02:20<00:08,  7.18it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 94%|█████████▍| 1003/1066 [02:20<00:10,  5.90it/s]

[{'generated_text': 'negative'}]
negative


 94%|█████████▍| 1004/1066 [02:20<00:12,  5.00it/s]

[{'generated_text': 'negative'}]
negative


 94%|█████████▍| 1006/1066 [02:21<00:12,  4.99it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 94%|█████████▍| 1007/1066 [02:21<00:13,  4.40it/s]

[{'generated_text': 'negative'}]
negative


 95%|█████████▍| 1008/1066 [02:21<00:14,  4.07it/s]

[{'generated_text': 'negative'}]
negative


 95%|█████████▍| 1010/1066 [02:22<00:12,  4.45it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 95%|█████████▍| 1011/1066 [02:22<00:11,  4.76it/s]

[{'generated_text': 'negative'}]
negative


 95%|█████████▍| 1012/1066 [02:22<00:11,  4.64it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 95%|█████████▌| 1014/1066 [02:22<00:08,  5.79it/s]

[{'generated_text': 'negative'}]
negative


 95%|█████████▌| 1015/1066 [02:23<00:10,  4.89it/s]

[{'generated_text': 'positive'}]
positive


 95%|█████████▌| 1017/1066 [02:23<00:09,  4.93it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 96%|█████████▌| 1019/1066 [02:23<00:07,  6.19it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 96%|█████████▌| 1022/1066 [02:24<00:05,  7.63it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 96%|█████████▌| 1024/1066 [02:24<00:05,  7.58it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 96%|█████████▌| 1026/1066 [02:24<00:04,  8.17it/s]

[{'generated_text': 'negative'}]
negative


 96%|█████████▋| 1028/1066 [02:24<00:05,  6.71it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 97%|█████████▋| 1030/1066 [02:25<00:05,  6.61it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 97%|█████████▋| 1032/1066 [02:25<00:05,  5.87it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive


 97%|█████████▋| 1034/1066 [02:25<00:05,  6.19it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 97%|█████████▋| 1036/1066 [02:26<00:05,  5.90it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 97%|█████████▋| 1039/1066 [02:26<00:03,  7.68it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'positive'}]
positive
[{'generated_text': 'positive'}]
positive


 98%|█████████▊| 1041/1066 [02:27<00:03,  6.68it/s]

[{'generated_text': 'positive'}]
positive
[{'generated_text': 'negative'}]
negative


 98%|█████████▊| 1042/1066 [02:27<00:03,  6.90it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 98%|█████████▊| 1045/1066 [02:27<00:02,  8.13it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 98%|█████████▊| 1048/1066 [02:27<00:01,  9.14it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 99%|█████████▊| 1052/1066 [02:28<00:01,  9.76it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 99%|█████████▉| 1054/1066 [02:28<00:01,  7.98it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 99%|█████████▉| 1055/1066 [02:28<00:01,  8.15it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 99%|█████████▉| 1057/1066 [02:28<00:01,  7.56it/s]

[{'generated_text': 'negative'}]
negative


 99%|█████████▉| 1059/1066 [02:29<00:01,  6.42it/s]

[{'generated_text': 'negative'}]
negative
[{'generated_text': 'negative'}]
negative


 99%|█████████▉| 1060/1066 [02:29<00:01,  5.05it/s]

[{'generated_text': 'negative'}]
negative


100%|█████████▉| 1061/1066 [02:29<00:01,  4.51it/s]

[{'generated_text': 'negative'}]
negative


100%|█████████▉| 1062/1066 [02:30<00:00,  4.29it/s]

[{'generated_text': 'negative'}]
negative


100%|█████████▉| 1063/1066 [02:30<00:00,  4.08it/s]

[{'generated_text': 'positive'}]
positive


100%|█████████▉| 1064/1066 [02:30<00:00,  3.70it/s]

[{'generated_text': 'negative'}]
negative


100%|█████████▉| 1065/1066 [02:31<00:00,  3.66it/s]

[{'generated_text': 'negative'}]
negative


100%|██████████| 1066/1066 [02:31<00:00,  7.05it/s]

[{'generated_text': 'negative'}]
negative


In [ ]:
evaluate_performance(data["test"]["label"], y_pred)

                 precision    recall  f1-score   support

Negative Review       0.83      0.85      0.84       533
Positive Review       0.85      0.83      0.84       533

       accuracy                           0.84      1066
      macro avg       0.84      0.84      0.84      1066
   weighted avg       0.84      0.84      0.84      1066



## using API's


In [ ]:
import openai
# Create client
client = openai.OpenAI(api_key="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")

In [ ]:
def chatgpt_generation(prompt, document, model="gpt-3.5-turbo-0125"):
  """Generate an output based on a prompt and an input document."""
  messages=[
    {
      "role": "system",
      "content": "You are a helpful assistant."
      },
    {

      "role": "user",
      "content": prompt.replace("[DOCUMENT]", document)
      }
  ]
  chat_completion = client.chat.completions.create(
  messages=messages,
  model=model,
  temperature=0
  )
  return
  chat_completion.choices[0].message.content

In [ ]:
# Define a prompt template as a base
prompt = """Predict whether the following document is a positive
or negative movie review:
[DOCUMENT]
If it is positive return 1 and if it is negative return 0. Do not
give any other answers.
"""
# Predict the target using GPT
document = "unpretentious , charming  , quirky , original"
chatgpt_generation(prompt, document)

In [ ]:

predictions = [
chatgpt_generation(prompt, doc) for doc in tqdm(data["test"]["text"])
]

In [ ]:
# Extract predictions
y_pred = [int(pred) for pred in predictions]


In [ ]:
# Evaluate performance
evaluate_performance(data["test"]["label"], y_pred)

In [ ]:
from google import genai
import os

os.environ["GOOGLE_API_KEY"] = "google-gemini-api"

client = genai.Client()

response = client.models.generate_content(
    model="gemini-3-flash-preview",
    contents="Explain how AI works in a few words",
)

print(response.text)

AI finds patterns in data to make predictions.


In [ ]:
def gemini_generation(prompt, document, model="gemini-3-flash-preview"):
  """Generate an output based on a prompt and an input document."""
  # Gemini API expects the prompt content directly
  full_prompt = prompt.replace("[DOCUMENT]", document)
  chat_completion = client.models.generate_content(
      model=model,
      contents=full_prompt,

  )
  return chat_completion.text

In [ ]:
# Define a prompt template as a base
prompt = """Predict whether the following document is a positive
or negative movie review:
[DOCUMENT]
If it is positive return 1 and if it is negative return 0. Do not
give any other answers.
"""
# Predict the target using GPT
document = "unpretentious , charming  , quirky , original"
gemini_generation(prompt, document)

'1'

In [ ]:
# You can skip this if you want to save your (free) credits
predictions = [gemini_generation(prompt, doc) for doc in tqdm(data["test"]["text"][0])]

  7%|▋         | 9/132 [01:15<17:10,  8.38s/it]


ClientError: 429 RESOURCE_EXHAUSTED. {'error': {'code': 429, 'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, head to: https://ai.google.dev/gemini-api/docs/rate-limits. To monitor your current usage, head to: https://ai.dev/usage?tab=rate-limit. \n* Quota exceeded for metric: generativelanguage.googleapis.com/generate_content_free_tier_requests, limit: 20, model: gemini-3-flash\nPlease retry in 5.100890611s.', 'status': 'RESOURCE_EXHAUSTED', 'details': [{'@type': 'type.googleapis.com/google.rpc.Help', 'links': [{'description': 'Learn more about Gemini API quotas', 'url': 'https://ai.google.dev/gemini-api/docs/rate-limits'}]}, {'@type': 'type.googleapis.com/google.rpc.QuotaFailure', 'violations': [{'quotaMetric': 'generativelanguage.googleapis.com/generate_content_free_tier_requests', 'quotaId': 'GenerateRequestsPerDayPerProjectPerModel-FreeTier', 'quotaDimensions': {'location': 'global', 'model': 'gemini-3-flash'}, 'quotaValue': '20'}]}, {'@type': 'type.googleapis.com/google.rpc.RetryInfo', 'retryDelay': '5s'}]}}

### Model Performance (F1 Scores)

| Approach | F1 Score |
| :--- | :--- |
| Task-Specific Model (RoBERTa) | 0.80 |
| Embeddings + Logistic Regression | 0.85 |
| Zero-Shot (Embeddings) | 0.78 |
| Flan-T5 (small) | 0.84 |
| ChatGPT (GPT-3.5) | 0.91 |
